### DISCUSSION TITLE
gum input as part of a shell pipe

### DISCUSSION THREAD

**Discussion by mil-ad** [charmbracelet/gum#643] [Q&A]
It seems like `gum input` is non-blocking so when I use it in a pipe like `gum input | echo blah` I immediately see `blah` printed before I'm done entering input into gum. Is this intended? How can I use `gum input` in a pipe and pass the output along?

**Comment by dev-nicolaos** [comment]
Ran into this and did some digging. Although its not directly using `gum input` in the pipeline I think many use cases can be covered by doing something like this:

```bash
foo=$(gum input) && echo $foo | my-command-that-reads-stdin
```

If you're concerned with leaving the extra variable in your environment you could append `&& unset foo` to the end of the line.

**Reply by evantaur** [reply]
You don't need a variable

redirect input:
`cowsay <<< $(gum input)`

Same but pipe to lolcat
`cowsay <<< $(gum input) | lolcat`
